icewm: 3.7.1 -> 3.7.3

Yarny0 opened the pull request:
Update IceWM to latest release.  Also drop a patch ([added here](https://github.com/NixOS/nixpkgs/pull/386156)) which is part of the new version, and [switch from `rev` to `tag` in `fetchFromGitHub`](https://github.com/NixOS/nixpkgs/pull/355973).

Release notes: https://github.com/ice-wm/icewm/releases/tag/3.7.3
Changes: https://github.com/ice-wm/icewm/compare/3.7.1...3.7.3


## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

More:

- [x] `nixosTests.{lightdm,sddm,xrdp}` build/pass (these seem to use IceWM in one way or another)
- [x] looks OK in qemu
- [x] I still can not reproduce the problem from https://github.com/NixOS/nixpkgs/issues/385959 , so it's solved upstream now

`nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` wants to rebuild 2k packages and was therefore skipped.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Yarny0 committed:
icewm: 3.7.1 -> 3.7.2

Release notes: https://github.com/ice-wm/icewm/releases/tag/3.7.2
Changes: https://github.com/ice-wm/icewm/compare/3.7.1...3.7.2

Also drops a patch which is now included in the new version, cf.
https://github.com/NixOS/nixpkgs/issues/385959
https://github.com/bbidulock/icewm/issues/794

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5361

Yarny0 committed:
icewm: 3.7.2 -> 3.7.3

Release notes: https://github.com/ice-wm/icewm/releases/tag/3.7.3
Changes: https://github.com/ice-wm/icewm/compare/3.7.2...3.7.3

Yarny0 commented:
Pushed update due to [new IceWM release](https://github.com/ice-wm/icewm/releases/tag/3.7.3).  Also rebased onto [current `nixos-unstable`](https://github.com/NixOS/nixpkgs/tree/42a1c966be226125b48c384171c44c651c236c22).  All "Things done" from PR text are still valid, the tests in particluar still pass.

Yarny0 committed:
icewm: replace `rev` with `tag` in `fetchFromGitHub`

Inspired by
https://github.com/NixOS/nixpkgs/commit/cb9f9a1e5a51a87d59b373db7016cee1608debd5

Yarny0 commented:
Rebased onto [current `nixos-unstable`](https://github.com/NixOS/nixpkgs/tree/c8cd81426f45942bb2906d5ed2fe21d2f19d95b7) to fix the recent merge conflict caused by https://github.com/NixOS/nixpkgs/pull/396749 and https://github.com/NixOS/nixpkgs/pull/397016 .  Inspired by those pull requests, I also added a commit that converts `rev` to `tag` just for IceWM.

All "Things done" from PR text are still valid, the tests in particluar still pass.  Also, I ensured that fetching via `tag` yields the same source hashes: I changed them and then compared the "got:" hash in the error message with the original one.

Yarny0 commented:
For the record: There is no merge conflict (according to Github).

I don't know how to deal with the "failing check":

> ofborg-eval: Commit not found

I hope that's not a blocker for this pull request.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-in-distress/3604/103

doronbehar commented:
A simple `git rebase` against latest `master` seem to have fixed the merge conflict and other ofborg errors.

Yarny0 commented:
Thanks, @doronbehar !
